packages/types/config/env.d.ts

```diff
@@ -3,4 +3,5 @@
  * Documentation: https://nuxtjs.org/api/configuration-env
  */
 
-export type NuxtConfigurationEnv = Record<string, string>
+export type NuxtConfigurationEnv = Record<string, any>
```

Comment:
Why `any` for the value. At least for real env variables, those can only be strings.

Reply:
Yes but you can make it resolved to whatever you want in configuration `env`, `publicRuntimeConfig` & `privateRuntimeConfig`.

It's especially useful for setting configuration boolean values, you don't want to check `value === '1' || value === 'true'` everywhere in your code base

Reply:
I guess the problem is I don't know what those new options are exactly. Are those documented somewhere?

I'm just saying that that change of types for normal ENVs is likely a breaking change (which I guess you are fine with but if possible, it would be better to avoid).

Reply:
Changing any type to `any` shouldn't be breaking change, you'll just lose intellisense like `string` methods, but the `any` type won't throw any error.

Those new options are related to a Nuxt 2.13 feature : https://github.com/nuxt/nuxt.js/pull/7312

Reply:
We're changing the value of env variables from `string` to `any` and that is a breaking change.

Reply:
@rchl That's not, `any` type means any type, which includes `string` ^^

Reply:
Also see : https://github.com/nuxt/nuxt.js/pull/7420

Reply:
Sorry, I'm not getting it.

Yes, `any` includes `string` but it's still can break existing code:

This will break:
```ts
const x: string = env.foo
````
I could come up with more realistic examples but you get the idea.

Reply:
I think having `NuctConfigurationRuntimeConfig` with `env` makes purposed misleading. runtimeConfig can be any arbitrary object using `process.env` at any level. The only relationship is that top-level values are expanded by env.

Reply:
Am I reading the linked change code correctly that `nuxt.env` will work as before and only expose native/string envs? It's the `nuxt.config` that will have the new ones? In that we shouldn't be changing types of `nuxt.env`.

@pi0 Maybe you can also shed some light here because I might be confused.

Reply:
> const x: string = env.foo

Alright I get your point

Well as people will use `$config` over `$nuxt.context.env`, changing type of `env` can bea voided

Reply:
Types should be better now, I made the PR only doing additional changes

Reply:
- Previous `config.env` is not touched and will be working as before but will be deprecated because it is implicitly shared between client and browser and not supporting dynamic envs
- `.env` the file is loaded as is (overridden by `process.env`) into an internal `config._env` object used to populate `process.env` and expand `runtimeConfig*` top-level keys. Only for auto expansion of `runtimeConfig*`, we use `destr` for better DX. So no types are broken